.circleci/config.yml

```diff
@@ -108,22 +176,74 @@ commands:
 
 jobs:
 
-  "build-and-test":
+  "build-dev-image":
+    executor: vm-linux
+    steps:
+      - checkout
+      - install-go
+      - run:
+          name: Build dev image
+          command: |
+            TAG="2.0.0-gotest.$CIRCLE_SHA1"
+            TELEPRESENCE_VERSION="v${TAG}" make image
+            mkdir -p /tmp/workspace
+            docker save "datawire/tel2:${TAG}" > /tmp/workspace/tel2-image.tar
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - tel2-image.tar
+
+  "build-and-test-windows":
+    executor:
+      name: win/default
+      shell: bash.exe
+    steps:
+      - checkout
+      - install-go
+      - install-kubectl
+      - run: choco install make
+      - install-sshfs-win
+      # - run: choco install rsync
+      - run:
+          name: Download wintun
+          command: |
+            mkdir -p build-output/bin
+            curl -L https://www.wintun.net/builds/wintun-0.12.zip | tar -vf - -Ox wintun/bin/amd64/wintun.dll > "build-output/bin/wintun.dll"
+      - attach_workspace:
+          at: "C:\\workspace"
+      - prepare-kluster:
+          platform: windows
+      - run:
+          command: |
+            DTEST_KUBECONFIG="$HOME/kubeconfig" PATH="/c/:/c/Program Files/SSHFS-Win/bin:$PATH" DTEST_REGISTRY="docker.io/datawire" make test
+          no_output_timeout: 25m
+      - cleanup-kluster:
+          platform: windows
+      - save-logs
+
+  "build-and-test-linux":
     executor: vm-linux
     steps:
       - checkout
       - install-go
       - run: make build
       - install-kubectl
       - install-sshfs
+      - attach_workspace:
+          at: /tmp/workspace
+      - prepare-kluster:
+          platform: linux
```

Comment:
Part of me doesn't like that we're doing this even when unnecessary (i.e. on GOOS=linux where dtest can spin up a local k3s cluster just fine).  But another part of me hates dtest and is excited that this gets us closer to killing it.